src/metrics/pipeline/applied/type.go

```diff
@@ -91,25 +91,25 @@ type OpUnion struct {
 
 // Equal determines whether two operation unions are equal.
 func (u OpUnion) Equal(other OpUnion) bool {
+	// keep in sync with Pipeline.Equal as go is terrible at inlining anything with a loop
 	if u.Type != other.Type {
 		return false
 	}
-	switch u.Type {
-	case pipeline.TransformationOpType:
-		return u.Transformation.Equal(other.Transformation)
-	case pipeline.RollupOpType:
+
+	if u.Type == pipeline.RollupOpType {
 		return u.Rollup.Equal(other.Rollup)
 	}
+
 	return true
 }
 
 // Clone clones an operation union.
 func (u OpUnion) Clone() OpUnion {
-	clone := OpUnion{Type: u.Type}
-	switch u.Type {
-	case pipeline.TransformationOpType:
-		clone.Transformation = u.Transformation.Clone()
-	case pipeline.RollupOpType:
+	clone := OpUnion{
+		Type:           u.Type,
+		Transformation: u.Transformation,
```

Comment:
does this need a clone as well? if so, two ifs is probably going to put you back in switch land branch wise

Reply:
nope, copying an int > copying the same single int conditionally